app/services/ZuoraService.scala

```diff
@@ -27,8 +30,69 @@ class ZuoraService(zuoraApiConfig: ZuoraApiConfig) extends ZuoraApi {
   override val metrics = new ZuoraMetrics(stage, application)
   val authTask = ScheduledTask(s"Zuora ${apiConfig.envName} auth", Authentication("", ""), 0.seconds, 30.minutes)(request(Login(apiConfig)))
 
-  def createSubscription(memberId: MemberId, data: SubscriptionData, ratePlanId: String, paymentDelay: Option[Period]): Future[SubscribeResult] = {
-    request(Subscribe(memberId, data, ratePlanId, paymentDelay))
+  def createSubscription(memberId: MemberId, data: SubscriptionData, paymentDelay: Option[Period]): Future[SubscribeResult] = {
+    request(Subscribe(memberId, data, paymentDelay))
   }
+
+  private def getProducts(): Future[Seq[SubscriptionProduct]] = {
+
+    def productRatePlans: Future[Seq[ProductRatePlan]] =
+      query[ProductRatePlan](s"ProductId='$digitalProductId'")
+
+
```

Comment:
Yeah maybe but I think it's a fair bit of work. We'd have to understand what queries we need to make to Zuora and then also how to extract it to something meaningful. I don't think this is too much overload but perhaps it will be when more products are introduced.